core/src/boot.py

```diff
@@ -3,14 +3,43 @@
 import storage.sd_salt
 from trezor import config, io, log, loop, res, ui, utils
 from trezor.pin import pin_to_int, show_pin_timeout
+from trezor.ui.confirm import HoldToConfirm
+from trezor.ui.text import Text
 
 from apps.common.request_pin import PinCancelled, request_pin
 from apps.common.sd_salt import SdProtectCancelled, request_sd_salt
 
 
+async def assert_sdcard() -> None:
+    sd = io.SDCard()
+    if not sd.present():
+        return
+
+    sd.power(True)
+    fs = io.FatFS()
+    try:
+        fs.mount()
+    except OSError:
+        text = Text("SD card", ui.ICON_WRONG, ui.RED)
+        text.normal("Unrecognized filesystem")
+        text.normal("found. Unplug the device")
+        text.normal("and remove the card or")
+        text.normal("you can format the card")
+        text.normal("losing all the data on it.")
```

Comment:
I would suggest something more like this:
```
text = Text("SD card error", ui.ICON_WRONG, ui.RED)
text.bold("Unknown filesystem.")
text.br_half()
text.normal("Use a different card or")
text.normal("you can format the card")
text.normal("losing all the data on it.")
```
1. We have started using bold + br_half to emphasize the gist of the error.
2. The user might not need to unplug the device, so "use" seems better.

I would also make the dialog look more like the one in apps/management/wipe_device.py (red button, flame icon). Better yet, I would split it into two dialogs:

SD card error
-
**Unknown filesystem.**

Use a different card or
format the SD card to
the FAT32 filesystem.
[Cancel]  [Format]

Format SD card
-
Do you really want to
format the SD card?

**All data on the SD card
will be lost.**
[Format SD card]